instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/internal/JettyHttpClient9TracingInterceptor.java

```diff
@@ -0,0 +1,154 @@
+    this.parentContext = parentCtx;
+    this.instrumenter = instrumenter;
+  }
+
+  public void attachToRequest(Request jettyRequest) {
+    List<JettyHttpClient9TracingInterceptor> current =
+        jettyRequest.getRequestListeners(JettyHttpClient9TracingInterceptor.class);
+
+    if (!current.isEmpty()) {
+      LOG.warn("A tracing interceptor is already in place for this request! ");
+      return;
+    }
+    startSpan(jettyRequest);
+
+    // wrap all important request-based listeners that may already be attached, null should ensure
+    // are returned here
+    List<Request.RequestListener> existingListeners = jettyRequest.getRequestListeners(null);
+    wrapRequestListeners(existingListeners);
+
+    jettyRequest
+        .onRequestBegin(this)
+        .onRequestFailure(this)
+        .onResponseFailure(this)
+        .onResponseSuccess(this);
+  }
+
+  private void wrapRequestListeners(List<Request.RequestListener> requestListeners) {
+
+    ListIterator<Request.RequestListener> iterator = requestListeners.listIterator();
+    while (iterator.hasNext()) {
+      Request.RequestListener requestListener = iterator.next();
+      if (requestListener instanceof Request.FailureListener) {
+        iterator.set(
+            (Request.FailureListener)
+                (request, throwable) -> {
+                  try (Scope ignore = context.makeCurrent()) {
+                    ((Request.FailureListener) requestListener).onFailure(request, throwable);
+                  }
+                });
+      }
+      if (requestListener instanceof Request.BeginListener) {
+        iterator.set(
+            (Request.FailureListener)
+                (request, throwable) -> {
+                  try (Scope ignore = context.makeCurrent()) {
+                    ((Request.BeginListener) requestListener).onBegin(request);
+                  }
+                });
+      }
+    }
+  }
+
+  private void startSpan(Request request) {
+
+    if (!instrumenter.shouldStart(this.parentContext, request)) {
+      return;
+    }
+    this.context = instrumenter.start(this.parentContext, request);
+  }
+
+  @Override
+  public void onBegin(Request request) {
+    if (this.context != null) {
+      Span span = Span.fromContext(this.context);
+      HttpField agentField = request.getHeaders().getField(HttpHeader.USER_AGENT);
+      span.setAttribute(SemanticAttributes.HTTP_USER_AGENT, agentField.getValue());
+    }
+  }
+
+  @Override
+  public void onComplete(Result result) {
+    closeIfPossible(result.getResponse());
+  }
+
+  @Override
+  public void onSuccess(Response response) {
+    closeIfPossible(response);
+  }
+
+  @Override
+  public void onFailure(Request request, Throwable t) {
+    if (this.context != null) {
+      instrumenter.end(this.context, request, null, t);
+    }
+  }
+
+  @Override
+  public void onFailure(Response response, Throwable t) {
+    if (this.context != null) {
+      instrumenter.end(this.context, null, response, t);
```

Comment:
```suggestion
      instrumenter.end(this.context, response.getRequest(), response, t);
```

Reply:
Looks like there was one more here, maybe this will clean up the attributes extractors

Reply:
Same could be done in the `closeIfPossible` method

Reply:
Works now if I pass the request, thanks!